apps/backend/src/api/configuration.py

```
@@ -0,0 +1,655 @@
+
+        # Calculate status statistics
+        total_errors = len(error_snapshots)
+        sync_errors = len([s for s in error_snapshots if s.sync_status == "error"])
+        validation_errors = len([s for s in error_snapshots if s.validation_status == "error"])
+
+        status_summary = {
+            "period_hours": hours,
+            "total_error_snapshots": total_errors,
+            "sync_errors": sync_errors,
+            "validation_errors": validation_errors,
+            "devices_with_errors": len(set(s.device_id for s in error_snapshots)),
```

Ruff (0.12.2)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)